src/components/pages/WelcomePage.tsx

```diff
@@ -763,19 +765,33 @@ class WelcomePage extends React.Component<Props, State> {
   private setAutocompleteOptions = async (path: string, input: string) => {
 
     // Handle place autocomplete
-    if (path === "/linkedevents/places/search") {
-      if (input.length < 3) {
-        return [];
-      }
-      const res = await fetch(`https://mantyharju-test.linkedevents.fi/v1/search/?type=place&input=${ input }`);
-      const data = await res.json();
-      return data.data.map((place: any) => {
-        return {
-          name: place.name && place.name.fi ? place.name.fi : place.id,
-          value: place.id
-        };
-      });
+    if (input.length < 1) {
+      return [];
+    }
+
+    if (Object.keys(this.state.fetchData).length === 0) {
+      const data = await this.fetchPlaces();
+      this.setState({
+        fetchData: data
+      })
     }
+
+    return await this.state.fetchData.data.map((place: any) => {
+      return {
+        name: place.name && place.name.fi ? place.name.fi : place.id,
+        value: place.id
+      };
+    });
```

Comment:
These should really be in separate functions. It is confusing that the function handles setting fetch data to state and also maps name value objects for autocomplete. Separating these would improve readability